Add strict_types=1 declaration

GaryJones opened the pull request:
## Description
On its own this doesn't do much, but when more type definitions are added, particularly scalar types, then PHP will handle the types without juggling them.

For choice of positioning below the file-level DocBlock, see https://www.php-fig.org/psr/psr-12/.
For lack of spaces inside the parentheses and around the `=`, see https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/.

## Motivation and Context
See #389.

## How Has This Been Tested?
Ran CI tests and quick visual inspection.

GaryJones committed:
Declare strict_types=1

On it's own this doesn't do much, but when more type definitions are added, particularly scalar types, then PHP will handle the types without juggling them.

For choice of positioning below the file-level DocBlock, see https://www.php-fig.org/psr/psr-12/.
For lack of spaces inside the parentheses and around the `=`, see https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/

See #389.

GaryJones commented:
Well, that caught a bug around types already! A boolean being passed to a `str_replace()` native PHP function, which produced a TypeError on PHP 8 and above.

I'll get this one fixed.

pauarge commented:
Looking good! Does it make sense to update our `CONTRIBUTING` or `README` files to let people know that we expect `strict_types=1` on all files?

GaryJones committed:
Fix PHP 8+ TypeError for getting current URL

> TypeError: str_replace(): Argument #3 ($subject) must be of type array|string, bool given

This originates when no `$post_id` is passed to the `get_current_url()` method, and no Post ID can be determined from the global `$post`.

As `get_permalink()` can return `false`, then casting it to a string means it will have a value of an empty string.

GaryJones committed:
Docs: Add contributor note about strong typing
